packages/popup/src/pages/ImportAccount/index.js

```diff
@@ -0,0 +1,166 @@
+  useEffect(() => {
+    setKeygenNamePlaceholder(`Seed-${keygenGroup.length + 1}`)
+  }, [keygenGroup])
+
+  const validateKeygen = keygen => {
+    setKeygenErrorMessage(keygen === '' ? 'Required!' : '')
+  }
+  const changeName = e => {
+    setName(e.target.value)
+  }
+  const changeKeygen = e => {
+    setKeygen(e.target.value)
+    validateKeygen(e.target.value)
+  }
+  const dispatchMutate = () => {
+    mutate([...GET_ALL_ACCOUNT_GROUP])
+    pattern === 'seed-phrase' && mutate([...GET_HD_ACCOUNT_GROUP])
+    pattern !== 'seed-phrase' && mutate([...GET_PK_ACCOUNT_GROUP])
+  }
+  const importGroup = () => {
+    let method = ''
+    let keygenType = ''
+    const params = {
+      password: createdPassword,
+      nickname: name || keygenNamePlaceholder,
+    }
+    if (pattern === 'private-key') {
+      method = 'wallet_importPrivateKey'
+      keygenType = 'privateKey'
+    } else {
+      method = 'wallet_importMnemonic'
+      keygenType = 'mnemonic'
+    }
+    params[keygenType] = keygen
+    return request(method, params)
+  }
+
+  const importAccount = async () => {
+    if (
+      !creatingAccount &&
+      name.length <= 20 &&
+      !keygenErrorMessage &&
+      keygen
+    ) {
+      setCreatingAccount(true)
+      try {
+        const res = await importGroup()
+        if (res?.error) {
+          throw res.error
+        }
+        if (res.result) {
+          setCreatingAccount(false)
+          dispatchMutate()
+          history.push('/')
+        }
+      } catch (err) {
+        setCreatingAccount(false)
+        err.message && setKeygenErrorMessage(err.message.split('\n')[0])
+      }
+    }
+  }
+
+  const onCreate = () => {
+    validateKeygen(keygen)
+    importAccount()
+  }
+
+  return (
+    <div className="bg-bg  h-full flex flex-col">
+      <TextNav hasGoBack={true} title={t(`${lanPrefix}Import`)} />
+      <main className="px-3 flex-1 flex flex-col">
+        <form
+          onSubmit={e => {
+            e.preventDefault()
+          }}
+          className="flex flex-col justify-between h-full"
+        >
+          <section>
+            <CompWithLabel label={t(`${lanPrefix}GroupName`)}>
+              <Input
+                onChange={changeName}
+                width="w-full"
+                placeholder={keygenNamePlaceholder}
+                maxLength="20"
+              />
+            </CompWithLabel>
+            <CompWithLabel
+              label={t(pattern === 'seed-phrase' ? 'seedPhrase' : 'pKey')}
+              labelStyle="mt-4 mb-3"
+            >
+              <Input
+                errorMessage={keygenErrorMessage}
+                elementType="textarea"
+                placeholder={t(`${lanPrefix}ImportPlaceholder`)}
+                onChange={changeKeygen}
+                width="w-full"
+                className="resize-none"
+                textareaSize="h-40"
+              ></Input>
+            </CompWithLabel>
+          </section>
+          <section className="h-14">
```

Comment:
should disable this button if the seed is invalid

Reply:
done